Community/Tdarr_Plugin_bsh1_Boosh_FFMPEG_QSV_HEVC.js

```diff
@@ -0,0 +1,400 @@
+  // (Entered means user actually wants something to happen, empty would disable this).
+  if (inputs.min_average_bitrate !== '') {
+    // Checks if inputs.bitrate_cutof is below inputs.min_average_bitrate.
+    // If so then set currentBitrate to the minimum allowed.)
+    if (inputs.bitrate_cutoff < inputs.min_average_bitrate) {
+      response.processFile = false;
+      response.infoLog += `☒ Bitrate cutoff ${inputs.bitrate_cutoff} is less than the set minimum 
+      average bitrate set of ${inputs.min_average_bitrate}. We don't want this. Cancelling plugin. \n`;
+      return response;
+    }
+    if (targetBitrate < inputs.min_average_bitrate) {
+      response.infoLog += `Target average bitrate clamped at min of ${inputs.min_average_bitrate}. \n`;
+      targetBitrate = Math.round(inputs.min_average_bitrate);
+      minimumBitrate = Math.round((targetBitrate * 0.75));
+      maximumBitrate = Math.round((targetBitrate * 1.25));
+    }
+  }
+
+  // Go through each stream in the file.
+  for (let i = 0; i < file.ffProbeData.streams.length; i += 1) {
+    // Check if stream is a video.
+    if (file.ffProbeData.streams[i].codec_type.toLowerCase() === 'video') {
+      // Check if codec of stream is mjpeg/png, if so then remove this "video" stream.
+      // mjpeg/png are usually embedded pictures that can cause havoc with plugins.
+      if (file.ffProbeData.streams[i].codec_name === 'mjpeg' || file.ffProbeData.streams[i].codec_name === 'png') {
+        extraArguments += `-map -v:${videoIdx} `;
+      }
+      // First check if we're reprocessing HEVC files, if not then we can ensure we don't convert HEVC again
+      if (inputs.reconvert_hevc !== 'true' && (file.ffProbeData.streams[i].codec_name === 'hevc'
+        || file.ffProbeData.streams[i].codec_name === 'vp9' || file.ffProbeData.streams[i].codec_name === 'av1')) {
+        // Check if codec of stream is hevc vp9, or av1 AND check if file.container matches inputs.container.
+        // If so nothing for plugin to do.
+        if ((file.ffProbeData.streams[i].codec_name === 'hevc' || file.ffProbeData.streams[i].codec_name === 'vp9'
+          || file.ffProbeData.streams[i].codec_name === 'av1') && file.container === inputs.container) {
+          response.processFile = false;
+          response.infoLog += `☑ File is already hevc or vp9 & in ${inputs.container}. \n`;
+          return response;
+        }
+        // Check if codec of stream is hevc, vp9 or av1
+        // AND check if file.container does NOT match inputs.container.
+        // If so remux file.
+        if ((file.ffProbeData.streams[i].codec_name === 'hevc' || file.ffProbeData.streams[i].codec_name === 'vp9'
+          || file.ffProbeData.streams[i].codec_name === 'av1') && file.container !== inputs.container) {
+          response.infoLog += `☒ File is hevc or vp9 but is not in ${inputs.container} container. Remuxing. \n`;
+          response.preset = `, -map 0 -c copy ${extraArguments}`;
```

Comment:
`<io>` over `,`

Reply:
This would probably need extra logic for streams that aren't compatible between mp4 and mkv, like `mov_text`

Reply:
Added in migz `force_conform` logic which should allow the encode to finish by dropping thigns like mov_text etc.